docs/user_config.md

```diff
@@ -0,0 +1,150 @@
+cfg = config.parse()
+config.set_enabled_resources(cfg.get('to-run', []))
+```
+
+### Run a defined set of services
+Your app has many services, call them A, B, C, and D. You have defined modes you
+want users to be able to start (e.g. consumer vs. enterprise) and you don't want
+them to have to remember which service is necessary for which mode.
+You want to be able to write a doc that tells users to run `tilt up consumer`
+the first time and know that's enough that they'll be set up. Running Tilt
+subsequent times without arguments should default to the mode they already have
+set.
+
+#### Command-Lines
+* `tilt up`: run all services
+* `tilt up consumer`: run only services in the "consumer" list
+#### Tiltfile
+```python
+config.define_string_list("to-run", args=True)
+cfg = config.parse()
+groups = {
+  'consumer': ['a', 'b', 'c'],
+  'enterprise': ['a', 'b', 'd'],
+}
+resources = []
+for arg in cfg.get('args', []):
+  if arg in groups:
+    resources += groups[arg]
+  else:
+    # also support specifying individual services instead of groups, e.g. `tilt up a b d`
+    resources.append(arg)
+config.set_enabled_resources(resources)
+```
+
+### Specify services to edit
+You have many services that your users frequently need to run, but don't expect
+to edit. Your YAML includes good enough images. (Perhaps they're from a suitably
+recent CI invocation, or refer to standard images.)
+#### Command-Lines
+* `tilt up`: run all services, editing none
+* `tilt up -- --to-edit b`: run all services, editing b (the '--' indicates the
+  end of `tilt up` options and the start of user-defined settings)
+* `tilt up -- consumer --to-edit b --to-edit c`: run consumer services and set
+  up b and c for editing.
+#### Tiltfile
+```python
+config.define_string_list("to-run", args=True)
+config.define_string_list("to-edit")
+cfg = config.parse()
+# omitting the code from above Tiltfile that includes correct resources
+# Only configure the build for services we want to edit
+# Thanks to Tilt's existing behavior, an image that is mentioned in YAML but has no build set
+# will use the existing tag.
+to_edit = cfg.get('to-edit', [])
+if 'a' in to_edit:
+  docker_build('a', './a')
+if 'b' in to_edit:
+  docker_build('b', './b')
+if 'c' in to_edit:
+  docker_build('c', './c')
+if 'd' in to_edit:
+  docker_build('d', './d')
+```
+
+## The Config File
+In addition to config settings coming from command-line args, Tilt will read
+them from a `tilt_config.json` in the same directory as the `Tiltfile`, if one
+exists. It should contain a single JSON dict, where the keys are config setting
+names and the values are their values (e.g., `{"to-edit": ["b", "c"]}`).
+As with any other file read by the Tiltfile, if this config file is changed
+while Tilt is running, Tilt will pick up that change and reexecute the Tiltfile.
+If the same setting is specified in both the config file and the command-line
+args, the value from the args takes precedent.
```

Comment:
```suggestion
args, the value from the args takes precedence.
```

Reply:
but also I don't know if we want to encourage using the config file for this sort of ephemeral/frequently changed arg?